config/mapflags_test.go

```diff
@@ -62,6 +65,66 @@ func Test_mapFlags_Set(t *testing.T) {
 						)
 					}
 				}
+				if len(m.values) > 0 {
+					if s := m.String(); s == "" {
+						t.Errorf("Failed to get string value for non empty mapFlag: %v", m.values)
+					}
+				}
+			}
+		})
+	}
+}
+
+func Test_mapFlags_Set_nil(t *testing.T) {
+	var m *mapFlags
+	if err := m.Set("foo"); err != nil || m != nil {
+		t.Error("Failed to get nil error or lazy init if m is nil")
+	}
+}
+
+func Test_mapFlags_UnmarshalYAML(t *testing.T) {
+	tests := []struct {
+		name    string
+		yml     string
+		wantErr bool
+		want    map[string]string
```

Comment:
I find it easier to write two table tests: one for invalid input with expected error and one for valid inputs testing for no error

Reply:
I thought I need more tests, so that's why it's a table.